tests/conftest.py

```diff
@@ -69,18 +69,22 @@ def pipx_temp_env_helper(pipx_shared_dir, tmp_path, monkeypatch, request, utils_
     bin_dir = Path(tmp_path) / "otherdir" / "pipxbindir"
     man_dir = Path(tmp_path) / "otherdir" / "pipxmandir"
 
-    monkeypatch.setattr(constants, "PIPX_SHARED_LIBS", pipx_shared_dir)
+    global_home_dir = Path(tmp_path) / "global" / "pipxhome"
+    global_bin_dir = Path(tmp_path) / "global_otherdir" / "pipxbindir"
+    global_man_dir = Path(tmp_path) / "global_otherdir" / "pipxmandir"
```

Comment:
This setup doesn't really test the behaviour in a realistic manner, IMHO. I wonder whether making use of [`pyfakefs`](https://pytest-pyfakefs.readthedocs.io/en/latest/) here would be worth the trouble?

Reply:
I am not sure why that's needed considering that the logic is still tested if the global directories instead of local directories are being used in tests? See e.g. 
https://github.com/pypa/pipx/blob/3e965b1d3c28fea1405d590e27f5095821a29a26/tests/test_environment.py#L60-L74
I can change `test_cli` function in this file to check if local paths are being used to contrast it with `test_cli_global` if that makes the things more robust.

Reply:
Done in [9597273](https://github.com/pypa/pipx/commit/959727300e42516968d1057cb374001e0be065a3)